pkg/intermediate/aggregate.go

```diff
@@ -54,12 +69,13 @@ func InitAggregationProcess(input AggregationInput) (*AggregationProcess, error)
 		return nil, fmt.Errorf("worker number cannot be <= 0")
 	}
 	return &AggregationProcess{
-		make(map[FlowKey][]entities.Record),
+		make(map[FlowKey]AggregationFlowRecord),
 		sync.RWMutex{},
 		input.MessageChan,
 		input.WorkerNum,
 		make([]*worker, 0),
 		input.CorrelateFields,
+		input.AggregateElements,
 		make(chan bool),
 	}, nil
```

Comment:
nit: it will be better to specify corresponding field names here for future maintenance.